go-controller/pkg/ovn/secondary_localnet_network_controller.go

```diff
@@ -353,11 +402,77 @@ func (oc *SecondaryLocalnetNetworkController) newRetryFramework(
 }
 
 func (oc *SecondaryLocalnetNetworkController) localnetPortNetworkNameOptions() map[string]string {
-	localnetLSPOptions := map[string]string{
-		"network_name": oc.GetNetworkName(),
+	return map[string]string{
+		"network_name": oc.findNetworkName(),
+	}
+}
+
+func (oc *SecondaryLocalnetNetworkController) checkBridgeMapping(obj interface{}) error {
+	if !config.OVNKubernetesFeature.EnableInterconnect {
+		return nil
+	}
+
+	pod, isPod := obj.(*corev1.Pod)
+	if !isPod {
+		return fmt.Errorf("could not cast %T object to *knet.Pod", obj)
+	}
+
+	if !util.PodScheduled(pod) {
+		return nil
+	}
+
+	if oc.nodeName != pod.Spec.NodeName {
+		return nil
+	}
+
+	knownNad, err := oc.podAttachedToKnownNAD(pod)
+	if err != nil {
+		return err
+	}
+	if !knownNad {
+		return nil
+	}
```

Comment:
Do we really need this ? if we are at this controller it means the pod has already a localnet NAD assigned.

Reply:
Ahh right, all good.

Reply:
btw if we can improve the controller to get events on specific pods only (i.e if they have some specific label or such), it can improve stuff in general, but this isnt the scope of this PR